src/buildercore/cloudformation.py

```diff
@@ -133,6 +133,12 @@ def read_template(stackname):
     output_fname = os.path.join(config.STACK_DIR, stackname + ".json")
     return json.load(open(output_fname, 'r'))
 
+def read_output(stackname, key):
+    data = core.describe_stack(stackname).meta.data # boto3
+    selected_outputs = [o for o in data['Outputs'] if o['OutputKey'] == key]
+    ensure(len(selected_outputs) == 1, "Too many outputs selected: %s" % selected_outputs)
```

Comment:
Could be replaced with `assert` unless `ensure` has some error logic?

Reply:
not fear, assert is actually stripped out under some conditions.

`assert` is meant more of a developer sanity check (which is why they may be stripped out), however the way we use them here is more of a formal check: we don't want the normal flow of execution to proceed if our assertion about the environment is false in development, or while testing or in production or in a compiled format etc.